contracts/MerkleDrop.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "hardhat/console.sol";
+import "./IDrop.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MerkleDrop is IDrop, Ownable {
+    using SafeERC20 for IERC20;
+
+    address public override token;
+    bytes32 public override merkleRoot;
+    bool public initialized;
+    uint256 public expireTimestamp;
+
+    mapping(address => bool) public claimed;
+
+    function init(
+        address owner_,
+        address token_,
+        bytes32 merkleRoot_,
+        uint256 expireTimestamp_
+    ) external {
+        require(!initialized, "Drop already Initialized");
+        initialized = true;
+        token = token_;
+        merkleRoot = merkleRoot_;
+        expireTimestamp = expireTimestamp_;
+        _transferOwnership(owner_);
+    }
+
+    function claim(
+        address account,
+        uint256 amount,
+        bytes32[] calldata merkleProof
+    ) external override {
+        require(!claimed[account], "Drop already claimed.");
+        bytes32 node = keccak256(abi.encodePacked(account, amount));
+        require(
+            MerkleProof.verify(merkleProof, merkleRoot, node),
+            "Invalid proof"
+        );
+        claimed[account] = true;
```

Comment:
We can also create and move to `isClaimed` function, so we can verify user has claimed or not independently by just calling the function

Reply:
Does this reduce any gas fees? or just for code standard?

Reply:
there will be no impact on gas, how will you check user is already claimed from UI?

Reply:
do we need to write two functions to set and get claims?

Reply:
only for get is enough